Source/FullScreenPokemon.ts

```diff
@@ -4829,6 +4829,64 @@ module FullScreenPokemon {
                 callback);
         }
 
+        /* partyActivate functions
+         */
+
+        /**
+         * Makes sure the Player has a Pokemon in the party that knows the necessary HM move.
+         * 
+         * @param player   The Player.
+         * @param thing   The Solid to be affected.
+         * @todo Eventually add check to make sure the Player beat the Gym leader needed to use the move.
+         */
+        checkPartyActivate(player: IPlayer, thing: IHMSolid): void {
+            var partyPokemon: IPokemon[] = player.FSP.ItemsHolder.getItem("PokemonInParty"),
+                moves: BattleMovr.IMove[],
+                move: IMoveSchema,
+                i: number,
+                j: number;
+
+            for (i = 0; i < partyPokemon.length; i += 1) {
+                moves = partyPokemon[i].moves;
+
+                for (j = 0; j < moves.length; j += 1) {
+                    if (moves[j].title === thing.moveName) {
+                        move = player.FSP.MathDecider.getConstant("moves")[moves[j].title];
+                        player.FSP.activateMove(move, player, partyPokemon[i]);
+                    }
+                }
+            }
+        }
+
+        /**
+         * Activates a Move.
+         * 
+         * @param move   The Move being activated.
+         * @param player   The Player.
+         * @param thing   The Solid to be affected.
+         * @todo Eventually add check to make sure the Player beat the Gym leader needed to use the move.
+         */
+        activateMove(move: IMoveSchema, player: IPlayer, pokemon: IPokemon): void {
+            move.partyActivate(player, pokemon);
+        }
+
+        /**
+         * Cuts a cuttable tree.
+         *
+         * @param player   The Player.
+         * @param pokemon   The Pokemon using Cut.
+         * @returns True if tree can be cut. 
+         * @todo Eventually add check to make sure the Player beat the Gym leader needed to use the move.
+         */
+        partyActivateCut(player: IPlayer, pokemon: IPokemon): boolean {
+            if (player.bordering[player.direction]) {
+                if (player.bordering[player.direction].title === "CuttableTree") {
```

Comment:
One of the two use cases needs it. If the player clicks A on a CuttableTree we already know that the thing in front of the Player is a CuttableTree. When the Player clicks on Cut in the menu partyActivateCut is called. At that point you aren't sure that there is a thing in front of the Player so I figured that I should have a check somewhere to make sure that the correct object is in front of the player.